tests/rbd_mirror/test_group_info_while_force_promote.py

```diff
@@ -0,0 +1,326 @@
+            if err:
+                raise Exception("Map, mount and run IOs failed for " + str(image_spec))
+            else:
+                log.info("Map, mount and IOs successful for " + str(image_spec))
+
+            # Create manual group mirror snapshot
+            out, err = rbd_primary.mirror.group.snapshot.add(
+                **{"group-spec": group_spec}
+            )
+            if err:
+                raise Exception("Failed to add manual mirror group snapshot %s", out)
+            snapshot_id = out.split(" ")[-1].rstrip()
+
+            # Wait for 50% sync to complete
+            status_spec = {
+                "pool": pool,
+                "group": pool_config.get("group"),
+                "format": "json",
+            }
+            if "namespace" in pool_config:
+                status_spec.update({"namespace": pool_config.get("namespace")})
+            else:
+                group_spec = pool + "/" + pool_config.get("group")
+
+            wait_till_image_sync_percent(rbd_primary, 50, **group_config)
+
+            """ Perform Force promote, group info and group snapshot list parallelly to verify
+            group info primary should be false and snapshot list should show incomplete snapshot """
+            with parallel() as p:
+                p.spawn(
+                    force_promote_secondary,
+                    group_spec,
+                    rbd_secondary=rbd_secondary,
+                )
+                p.spawn(
+                    check_group_snap_list,
+                    snapshot_id,
+                    verify_state="incomplete",
+                    rbd_secondary=rbd_secondary,
+                    **status_spec,
+                )
+                p.spawn(
+                    check_group_info,
+                    expected_primary_field=False,
+                    rbd_secondary=rbd_secondary,
+                    **group_config,
+                )
+
+            # Wait for force promote to complete
+            time.sleep(10)
+
+            # Check group snap list on site-b, the snapshot should be marked complete
+            snapshot_id = get_mirror_group_snap_id(rbd_secondary, **status_spec)
+            check_group_snap_list(
+                snapshot_id, "complete", rbd_secondary=rbd_secondary, **status_spec
+            )
+
+            # Check group info, 'primary' field should be marked true
+            check_group_info(
+                expected_primary_field=True, rbd_secondary=rbd_secondary, **group_config
+            )
+
+
+def force_promote_secondary(group_spec, rbd_secondary):
+    (out, err) = rbd_secondary.mirror.group.promote(
+        **{"group-spec": group_spec, "force": True}
+    )
+    if err:
+        raise Exception("Failed to force promote group on site-B: " + str(err))
+    else:
+        log.info("Successfully completed force promote on secondary")
+
+
+def check_group_snap_list(snapshot_id, verify_state, rbd_secondary, **status_spec):
```

Comment:
let's add this in group workflow folder so that it can be reused for other tests